src/main/java/org/mondemand/transport/LWESTransport.java

```diff
@@ -335,20 +337,72 @@ public void sendTrace (String programId,
           traceMsg.setString(contexts[i].getKey(), contexts[i].getValue());
       }
       // emit the event
-      emitter.emit(traceMsg);
+      emitterGroup.emitToGroup(traceMsg);
     } catch(Exception e) {
       throw new TransportException("Error sending log event", e);
     }
   }
 
+  private static final String PERF_ID_KEY = "id";
+  private static final String PERF_CALLER_LABEL_KEY = "caller_label";
+  private static final String PERF_NUM_KEY = "num";
+  private static final String PERF_LABEL_PREFIX = "label";
+  private static final String PERF_START_PREFIX = "start";
+  private static final String PERF_END_PREFIX = "end";
+
+  public void sendPerformanceTrace(String id, String callerLabel,
+                                   String[] label, long[] start,
+                                   long[] end, Context[] contexts)
+    throws TransportException
+  {
+    if (id == null || callerLabel == null || label == null || start == null ||
+        end == null || contexts == null)
+    {
+      return;
+    }
+
+    if (label.length != start.length || label.length != end.length)
+    {
+      return;
```

Comment:
yeah, i was following the other send functions but will update this one at least.